tests/src/services/users.service.test.ts

```diff
@@ -0,0 +1,235 @@
+      const changeStorageSpy = jest.spyOn(storageService, 'changeStorage').mockImplementation(voidPromise);
+
+      await paymentService.getActiveSubscriptions(mocks.mockedUser.customerId);
+
+      await usersService.cancelUserIndividualSubscriptions(mocks.mockedUser.customerId);
+      await storageService.changeStorage(mocks.mockedUser.uuid, FREE_PLAN_BYTES_SPACE);
+
+      const individualSubscriptions = mocks.mockActiveSubscriptions.filter(
+        (sub) => sub.product?.metadata.type !== 'business',
+      );
+      expect(cancelSubscriptionSpy).toHaveBeenCalledTimes(individualSubscriptions.length);
+
+      expect(changeStorageSpy).toHaveBeenCalledTimes(1);
+      expect(changeStorageSpy).toHaveBeenCalledWith(mocks.mockedUser.uuid, FREE_PLAN_BYTES_SPACE);
+    });
+
+    it('Cancel the user B2B subscription', async () => {
+      jest
+        .spyOn(paymentService, 'getActiveSubscriptions')
+        .mockImplementation(() => Promise.resolve(mocks.mockActiveSubscriptions as unknown as ExtendedSubscription[]));
+
+      const cancelSubscriptionSpy = jest.spyOn(paymentService, 'cancelSubscription').mockImplementation(voidPromise);
+
+      const changeStorageSpy = jest.spyOn(storageService, 'changeStorage').mockImplementation(voidPromise);
+
+      await usersService.cancelUserB2BSuscriptions(mocks.mockedUser.customerId);
+      await storageService.changeStorage(mocks.mockedUser.uuid, FREE_PLAN_BYTES_SPACE);
+
+      const b2bSubscriptions = mocks.mockActiveSubscriptions.filter((sub) => sub.product?.metadata.type === 'business');
+
+      expect(cancelSubscriptionSpy).toHaveBeenCalledTimes(b2bSubscriptions.length);
+
+      b2bSubscriptions.forEach((sub) => {
+        expect(cancelSubscriptionSpy).toHaveBeenCalledWith(sub.id);
+      });
+
+      expect(changeStorageSpy).toHaveBeenCalledTimes(1);
+      expect(changeStorageSpy).toHaveBeenCalledWith(mocks.mockedUser.uuid, FREE_PLAN_BYTES_SPACE);
+    });
+  });
+
+  describe('Storing coupon user by user', () => {
+    it('should store the coupon successfully when the coupon is tracked', async () => {
+      (couponsRepository.findByCode as jest.Mock).mockResolvedValue(mocks.mockedCoupon);
+
+      await usersService.storeCouponUsedByUser(mocks.mockedUser, mocks.mockedCoupon.code);
+
+      expect(couponsRepository.findByCode).toHaveBeenCalledWith(mocks.mockedCoupon.code);
+      expect(usersCouponsRepository.create).toHaveBeenCalledWith({
+        coupon: mocks.mockedCoupon.id,
+        user: mocks.mockedUser.id,
+      });
+    });
+
+    it('should throw CouponNotBeingTrackedError when the coupon is not tracked', async () => {
+      (couponsRepository.findByCode as jest.Mock).mockResolvedValue(null);
+
+      await expect(usersService.storeCouponUsedByUser(mocks.mockedUser, 'INVALID_COUPON')).rejects.toThrow(
+        CouponNotBeingTrackedError,
+      );
+
+      expect(couponsRepository.findByCode).toHaveBeenCalledWith('INVALID_COUPON');
```

Comment:
Storing the `INVALID_COUPON` string in a constant is better for test maintenance purposes